theano/gof/params_type.py

```diff
@@ -611,11 +647,13 @@ def c_support_code(self):
             }
         };
         #endif
-        """ % dict(support_code=support_code,
-                   struct_name_defined=struct_name_defined, struct_name=struct_name, struct_declare=struct_declare,
+        /** End ParamsType %(struct_name)s **/
+        """ % dict(struct_name_defined=struct_name_defined, struct_name=struct_name, struct_declare=struct_declare,
                    struct_init=struct_init, struct_cleanup=struct_cleanup, struct_extract=struct_extract,
                    struct_extract_method=struct_extract_method)
 
+        return list(sorted(list(c_support_code_set))) + [final_struct_code]
+
     def c_code_cache_version(self):
         return ((1, 8), tuple(t.c_code_cache_version() for t in self.types))
```

Comment:
You probably have to update the cache version number here, since the definition of `c_support_code` has changed.